### PR TITLE
chore(Enumeration): disabling enumeration

### PR DESCRIPTION
**Pull request by jlame**
**What is the problem this PR is trying to solve?**
there's no disable state for enumeration widget

**What is the chosen solution to this problem?**
adding what's necessary to make it work

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Enumeration/Header/HeaderInput.component.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ function getAction(action, index, getInternalInputRef) {
 		}
 	}
 
-	return (
+	return !action.disabled ? (
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/HeaderInput.component.js]: you can add a test on disabled and if it is the case return null.
add if(action.disabled) { return null }

File: packages/components/src/Enumeration/Header/Header.component.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function getAction(action, index) {
 	}
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/Header.component.js]: you can add a test on disabled and if it is the case return null.
add if(action.disabled) { return null }

File: packages/components/src/Enumeration/Header/Header.component.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ function getAction(action, index) {
...
 
 	if (action.displayMode === 'dropdown') {
-		return (
+		return !action.disabled ? (
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/Header.component.js]: with the previous test it is not necessary.

File: packages/components/src/Enumeration/Header/Header.component.js [hunk 4]
```diff
@@ -31,9 +31,9 @@ function getAction(action, index) {
 				pullRight
 				link
 			/>
-		);
+		) : null;
 	}
-	return (
+	return !action.disabled ? (
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/Header.component.js]: idem

File: packages/components/src/Enumeration/Header/HeaderSelected.component.js [hunk 5]
```diff
@@ -18,7 +18,7 @@ function getAction(action, index) {
 		}
 	}
 
-	return (
+	return !action.disabled ? (
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Header/HeaderSelected.component.js]: you can add a test on disabled and if it is the case return null.
add if(action.disabled) { return null }

File: packages/components/src/Enumeration/Items/Item/Item.component.js [hunk 6]
```diff
@@ -48,7 +48,7 @@ function Item({ id, item, searchCriteria, showCheckboxes, style, t }) {
 			}
 		}
 
-		return (
+		return !action.disabled ? (
```
Comment (jare-talend) [inline packages/components/src/Enumeration/Items/Item/Item.component.js]: you can add a test on disabled and if it is the case return null.
add if(action.disabled) { return null }

File: packages/components/src/Enumeration/Header/HeaderInput.component.js (hunk 1 above)
Comment (jlame) [reply packages/components/src/Enumeration/Header/HeaderInput.component.js]: test already done

File: packages/components/src/Enumeration/Header/Header.component.js (hunk 2 above)
Comment (jlame) [reply packages/components/src/Enumeration/Header/Header.component.js]: test already done

File: packages/components/src/Enumeration/Header/HeaderSelected.component.js (hunk 5 above)
Comment (jlame) [reply packages/components/src/Enumeration/Header/HeaderSelected.component.js]: test already done

File: packages/forms/stories/json/enumerationDisabled.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  "jsonSchema": {
+    "title": "A enumeration form",
+    "description": "A enumeration form example.",
+    "type": "object",
+    "properties": {
+      "documents": {
+        "type": "array",
+        "required": true,
+        "allowDuplicates": true,
+        "disabled": true,
```
Comment (jsomsanith) [inline packages/forms/stories/json/enumerationDisabled.json]: This should be in uiSchema

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 8]
```diff
@@ -70,6 +70,7 @@ class EnumerationForm extends React.Component {
 		this.timerSearch = null;
 		this.allowDuplicate = false;
 		this.allowImport = false;
+		this.disabledAction = props.schema ? props.schema.disabled : false;
```
Comment (jsomsanith) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: No need to store it in the instance. You only use it in constructor. Use a const.
```
const 
disabled = props.schema && props.schema.disabled;
```

File: packages/components/src/Enumeration/Header/Header.component.js [hunk 9]
```diff
@@ -18,7 +18,10 @@ function getAction(action, index) {
 			action.onClick(event, { value: event.target.value });
 		}
 	}
-
+	if (action.disabled) {
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Header/Header.component.js]: You should filter the actions in the Header `return`, before the map
```
{
    headerDefault
        .filter(...)
        .map(getAction);
}
```

File: packages/components/src/Enumeration/Items/Item/Item.test.js [hunk 10]
```diff
@@ -64,6 +64,24 @@ describe('Item', () => {
 		expect(props.item.itemProps.onSelectItem).toBeCalled();
 	});
 
+	it('should display value with only button which are not disabled', () => {
+		// given
+		const itemWithDisabled = { ...item };
+		itemWithDisabled.itemProps.actions[0].disabled = true;
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Items/Item/Item.test.js]: You change the action from the same reference as all the tests.
Please don't mutate it as it can impact all the following tests. --> deepClone.

File: packages/components/src/Enumeration/Items/Item/Item.test.js [hunk 11]
```diff
@@ -64,6 +64,24 @@ describe('Item', () => {
...
+		};
+
+		const itemInstance = <Item {...props} />;
+
+		// when
+		const wrapper = mount(itemInstance);
+		const buttons = wrapper.find(Button);
+
+		// then
+		expect(buttons.length).toBe(2);
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Items/Item/Item.test.js]: I don't understand the assertion. You have 2 actions, 1 disabled, tou should have only 1 button

File: packages/components/src/Enumeration/Header/HeaderInput.component.js [hunk 12]
```diff
@@ -24,7 +24,9 @@ function getAction(action, index, getInternalInputRef) {
 			});
 		}
 	}
-
+	if (action.disabled) {
+		return null;
+	}
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Header/HeaderInput.component.js]: Same remark  : You should filter the actions in the main component `return`, before the map()

File: packages/components/src/Enumeration/Header/HeaderSelected.component.js [hunk 13]
```diff
@@ -17,7 +17,9 @@ function getAction(action, index) {
 			action.onClick(event);
 		}
 	}
-
+	if (action.disabled) {
+		return null;
+	}
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Header/HeaderSelected.component.js]: Same remark : You should filter the actions in the main component return, before the map()

File: packages/components/src/Enumeration/Items/Item/Item.component.js [hunk 14]
```diff
@@ -47,7 +47,9 @@ function Item({ id, item, searchCriteria, showCheckboxes, style, t }) {
 				});
 			}
 		}
-
+		if (action.disabled) {
+			return null;
+		}
```
Comment (jsomsanith) [inline packages/components/src/Enumeration/Items/Item/Item.component.js]: Same remark : You should filter the actions in the main component return, before the map()

File: packages/components/src/Enumeration/Items/Item/Item.test.js (hunk 11 above)
Comment (jlame) [reply packages/components/src/Enumeration/Items/Item/Item.test.js]: the selection also includes the title which is clickable and is indeed a button, i've narrow the selection so we only check the one we're interested in.
